scheduler/task_priority_cmp.go

```diff
@@ -115,39 +117,36 @@ func tasksAreCommitBuilds(t1, t2 task.Task) bool {
 // list. This is to ensure that task groups are dispatched in the order that
 // they are defined.
 func byTaskGroupOrder(t1, t2 task.Task, comparator *CmpBasedTaskComparator) (int, error) {
-	if t1.Version != t2.Version {
-		return 0, nil
-	}
-
+	// Try other comparators if both tasks are not in task groups
 	if t1.TaskGroup == "" && t2.TaskGroup == "" {
 		return 0, nil
 	}
 
+	// If one task is in a task group, sort that one higher, which keeps the pre-byTaskGroupOrder order.
 	if t2.TaskGroup == "" && t1.TaskGroup != "" {
-		return 0, nil
+		return 1, nil
 	}
-
 	if t1.TaskGroup == "" && t2.TaskGroup != "" {
 		return -1, nil
 	}
 
-	if t1.TaskGroup != t2.TaskGroup {
-		return 1, nil
-	}
-
-	if t1.BuildId != t2.BuildId {
-		return 1, nil
-	}
-
-	if t1.TaskGroupOrder > t2.TaskGroupOrder {
-		return -1, nil
+	// If tasks are in the same task group and build, apply the task group comparator.
+	if t1.TaskGroup == t2.TaskGroup && t1.BuildId == t2.BuildId {
+		if t1.TaskGroupOrder > t2.TaskGroupOrder {
```

Comment:
A more succinct (though less clear) way is to always return t2.order - t1.order

Reply:
that seems less desirable.